rust/connlib/snownet/tests/lib.rs

```diff
@@ -855,3 +924,90 @@ fn progress(
         }
     }
 }
+
+fn icmp_request_packet(source: IpAddr, dst: IpAddr) -> MutableIpPacket<'static> {
+    match (source, dst) {
+        (IpAddr::V4(src), IpAddr::V4(dst)) => {
+            use pnet_packet::{
+                icmp::{
+                    echo_request::{IcmpCodes, MutableEchoRequestPacket},
+                    IcmpTypes, MutableIcmpPacket,
+                },
+                ip::IpNextHeaderProtocols,
+                ipv4::MutableIpv4Packet,
+                MutablePacket as _, Packet as _,
+            };
+
+            let mut buf = vec![0u8; 60];
+
+            let mut ipv4_packet = MutableIpv4Packet::new(&mut buf[..]).unwrap();
+            ipv4_packet.set_version(4);
```

Comment:
I found for tests it's a bit easier to read if we use [populate](https://docs.rs/pnet/latest/pnet/packet/ipv4/struct.MutableIpv4Packet.html#method.populate) but I don't have a strong opinion

Reply:
I want to extract our own `ip-packet` crate first and then we can make this nicer :)